src/Symfony/Component/Routing/Matcher/Dumper/StaticUrlMatcherDumper.php

```diff
@@ -0,0 +1,296 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Routing\Matcher\Dumper;
+
+use Symfony\Component\Routing\Route;
+use Symfony\Component\Routing\RouteCollection;
+use Symfony\Component\ExpressionLanguage\ExpressionLanguage;
+use Symfony\Component\ExpressionLanguage\ExpressionFunctionProviderInterface;
+
+/**
+ * StaticUrlMatcherDumper creates a PHP array to be used with StaticUrlMatcher.
+ *
+ * @author Fabien Potencier <fabien@symfony.com>
+ * @author Tobias Schultze <http://tobion.de>
+ * @author Arnaud Le Blanc <arnaud.lb@gmail.com>
+ * @author Nicolas Grekas <p@tchwork.com>
+ */
+class StaticUrlMatcherDumper extends MatcherDumper
+{
+    private $expressionLanguage;
+    private $expressions = array();
+    private $expressionsIndex = array();
+
+    /**
+     * @var ExpressionFunctionProviderInterface[]
+     */
+    private $expressionLanguageProviders = array();
+
+    /**
+     * {@inheritdoc}
+     */
+    public function dump(array $options = array())
+    {
+        list($routes, $expressions) = $this->compileRoutes($this->getRoutes());
+
+        return <<<EOF
+<?php
+
+// This file has been auto-generated by the Symfony Routing Component.
+
+return array(
+    {$this->exportRoutes($routes)},
+    {$this->exportExpressions($expressions)},
+);
+
+EOF;
+    }
+
+    public function addExpressionLanguageProvider(ExpressionFunctionProviderInterface $provider)
+    {
+        $this->expressionLanguageProviders[] = $provider;
+    }
+
+    private function compileRoutes(RouteCollection $routes): array
```

Comment:
@alexpott you might be interested into having this method public at some point for Drupal, as it allows building a static routing table that can be (json)serialized and used together with `StaticUrlMatcher`

Reply:
Right now our entire routing system uses a database tables. This was introduced due to some reasons:

* There might be a lot of possible routes
* User changes might trigger new routes

We then leverage the DB to do our filtering, instead of something purely regex based. Does that make sense for you?

We explored a bit the idea of doing some static optimizations for like the frontpage or so, but quickly realized that there is too much flexibility in the existing system and more important, the bottleneck is not the routing system at all.

Reply:
yep, makes sense, yet I don't know if it's actually better to model routes in the DB (some rows per routes, which is what you do now?), vs just putting one json that describes the full routing in the DB, which could be generated here. I don't have the answer, I just wanted to let you know :)